akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -147,11 +152,12 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
   }
 
   private final class SnapshotFilenameFilter(persistenceId: String) extends FilenameFilter {
+    val encodedPersistenceId = URLEncoder.encode(persistenceId)
+
     def accept(dir: File, name: String): Boolean = {
-      name match {
-        case FilenamePattern(pid, snr, tms) ⇒ pid.equals(URLEncoder.encode(persistenceId))
-        case _                              ⇒ false
-      }
+      val persistenceIdEndIdx = name.lastIndexOf('-', name.lastIndexOf('-') - 1)
```

Comment:
looks like it would be easy to add some unit testing of the `SnapshotFilenameFilter` class, e.g. testing names where the persistenceId includes a `-`